Testscripts/Linux/channel_change.sh

```diff
@@ -19,229 +13,160 @@
-						sleep 1
-						newState=$(cat /sys/devices/system/cpu/cpu$_cpu_id/online)
-						# Verify the cpu id change
-						if [[ $newState == $oldState ]]; then
-							LogMsg "Successfully verified no cpu state change"
-						else
-							LogErr "Failed to verify the cpu state change. Expected $oldState, found $newState"
-						fi
+	# restore the original vmbus and cpu id restored in OriginalSource file.
+	OriginalSource=vmbus_cpu.original
+	if [ -f $OriginalSource ]; then
+		rm -f $OriginalSource
+	fi
+	touch $OriginalSource
+	LogMsg "Changed all vmbus channels' cpu id to 0, if non-zero"
+	for _device in /sys/bus/vmbus/devices/*
+	do
+		echo $_device >> $OriginalSource
+		# Need to access the copy of channel_vp_mapping to avoid race condition.
+		cp $_device/channel_vp_mapping .
+		# read channel_vp_mapping file of each device
+		while IFS=: read _vmbus_ch _cpu
+		do
+			LogMsg "vmbus: $_vmbus_ch,		cpu id: $_cpu"
+			echo "$_vmbus_ch:$_cpu" >> $OriginalSource
+			if [ $_cpu != 0 ]; then
+				idle_cpus+=($_cpu)
+				# Set 0 to online file, echo 0 > /sys/devices/system/cpu/cpu<number>/online
+				oldState=$(cat /sys/devices/system/cpu/cpu$_cpu/online)
+				echo 0 > /sys/devices/system/cpu/cpu$_cpu/online 2>&1
+				sleep 1
+				newState=$(cat /sys/devices/system/cpu/cpu$_cpu/online)
+				# Verify the cpu id change
+				if [[ $newState = $oldState ]]; then
+					LogMsg "Successfully verified NO cpu state change, because it was used."
+				else
+					LogErr "Failed to verify the cpu state NO change. Expected $oldState, found $newState"
+					FailedCount=$((FailedCount+1))
+				fi
 
-						# Now reset this vmbus's cpu id to 0. The target cpu id is ready to be idle.
-						LogMsg "Found the cpu id, $_cpu_id from the channel $rel_id. Will set 0, the default cpu id, to this cpu"
-						echo 0 > $sysfs_path/channels/$rel_id/cpu
-						sleep 1
-						# validate the change of that cpu id
-						_cpu_id=$(cat $sysfs_path/channels/$rel_id/cpu)
-						if [[ $_cpu_id == "0" ]]; then
-							LogMsg "Successfully changed the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0"
-							cpu_idx=$((cpu_idx+1))
-						else
-							LogErr "Failed to change the cpu id of the channel $rel_id from ${idle_cpus[$cpu_idx]} to 0. Expected 0, but found $_cpu_id"
-							FailedCount=$((FailedCount+1))
-						fi
-					fi
-					read -a line
-				done
-				#LogMsg "Found vmbus channel and its target cpus: ${temp[@]}, ${!temp[@]}"
-			else
-				LogErr "Failed. Supposed to read Sysfs path, but found ${line[@]}"
-				FailedCount=$((FailedCount+1))
+				# Now reset this cpu id to 0.
+				LogMsg "Set the vmbus channel $_vmbus_ch's cpu to the default cpu, 0"
+				echo 0 > $_device/channels/$_vmbus_ch/cpu
+				sleep 1
+				# validate the change of that cpu id
+				_cpu_id=$(cat $_device/channels/$_vmbus_ch/cpu)
+				if [ $_cpu_id = 0 ]; then
+					LogMsg "Successfully set the vmbus channel $_vmbus_ch's cpu to 0"
+					cpu_idx=$((cpu_idx+1))
```

Comment:
"cpu_idx" is only used here, IIUC, the line can be removed

Reply:
Fixed in 6b1643883a8b1ad8eb082a0e161c107e1bdd9efb